src/components/MapLayerList/Layer.js

```diff
@@ -1,27 +1,33 @@
-/*
-Purpose
-  individual layer controls vissibility of layer on the map
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Checkbox } from '@mui/material';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import { useDispatch, useSelector } from 'react-redux';
+import { toggleLayer } from '../../reducers/mapLayerListSlice';
+import LayerDescription from './LayerDescription';
+import LayerLegend from './LayerLegend';
 
-Child Components
-  - maybe map.js
-  - MapLayerList-LayerLegend.js
-  - MapLayerList-LayerDescription.js
+export default function Layer(props) {
+  const { lData } = props;
+  const dispatch = useDispatch();
+  const layerListSelector = (state) => state.mapLayerList.activeLayerList;
+  const activeLayerList = useSelector(layerListSelector);
+  const checked = lData.id in activeLayerList;
 
-Libs
-  - leaflet
+  const handleClick = () => {
+    dispatch(toggleLayer(lData));
+  };
 
-API
-  - Not sure yet
+  return (
+    <AccordionDetails>
+      <Checkbox checked={checked} onClick={() => handleClick()} />
+      {lData.label}
+      <LayerLegend layer={lData} />
+      <LayerDescription layerName={lData.label} layerDescription={lData.description} />
+    </AccordionDetails>
+  );
+}
 
-State needed
-  - layer vissibility
-  - legend vissibility
-
-Props
-  - layer name
-  - layer Description
-  - layer legend
-  - layer vissibility
-  - layer legend vissibility
-  - Not sure yet
-*/
```

Comment:
Did you mean to delete all explanation comments?

Reply:
These are removed now